### ISSUE TITLE
pod install --repo-update doesn't work for AppLovin example project

### ISSUE THREAD

**Issue by Gargo** [CocoaPods/CocoaPods#11814]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

### Command

```
/opt/homebrew/Cellar/cocoapods/1.11.3_1/libexec/bin/pod install --repo-update
```

### Report

* What did you do?

Downloaded AppLovin example:
https://github.com/AppLovin/AppLovin-MAX-SDK-iOS
and run command:
```
pod install --repo-update
```

* What did you expect to happen?

no error

* What happened instead?
error with detailed info:

### Stack

```
   CocoaPods : 1.12.0
        Ruby : ruby 3.2.1 (2023-02-08 revision 31819e82c8) [arm64-darwin21]
    RubyGems : 3.4.6
        Host : macOS 12.5 (21G72)
       Xcode : 14.2 (14C18)
         Git : git version 2.37.1 (Apple Git-137.1)
Ruby lib dir : /opt/homebrew/Cellar/ruby/3.2.1/lib
Repositories : trunk - CDN - https://cdn.cocoapods.org/
```

### Plugins

```
cocoapods-deintegrate : 1.0.5
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.1
cocoapods-trunk       : 1.6.0
cocoapods-try         : 1.2.0
```

### Podfile

```ruby
use_frameworks!
inhibit_all_warnings!
platform :ios, '10.0'

target 'AppLovin MAX Demo App - Swift' do
  pod 'AppLovinSDK'
  pod 'Adjust'
end
```

### Error

```
LoadError - linked to incompatible /Users/admin/.rvm/rubies/ruby-3.2.1/lib/libruby.3.2.dylib - /Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/ffi-1.15.5/lib/ffi_c.bundle
<internal:/opt/homebrew/Cellar/ruby/3.2.1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:88:in `require'
<internal:/opt/homebrew/Cellar/ruby/3.2.1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:88:in `require'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/ffi-1.15.5/lib/ffi.rb:5:in `rescue in <top (required)>'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/ffi-1.15.5/lib/ffi.rb:2:in `<top (required)>'
<internal:/opt/homebrew/Cellar/ruby/3.2.1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:88:in `require'
<internal:/opt/homebrew/Cellar/ruby/3.2.1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:88:in `require'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/ethon-0.16.0/lib/ethon.rb:3:in `<top (required)>'
<internal:/opt/homebrew/Cellar/ruby/3.2.1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:88:in `require'
<internal:/opt/homebrew/Cellar/ruby/3.2.1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:88:in `require'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/typhoeus-1.4.0/lib/typhoeus.rb:2:in `<top (required)>'
<internal:/opt/homebrew/Cellar/ruby/3.2.1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:88:in `require'
<internal:/opt/homebrew/Cellar/ruby/3.2.1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:88:in `require'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-core-1.12.0/lib/cocoapods-core/cdn_source.rb:440:in `download_typhoeus_impl_async'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-core-1.12.0/lib/cocoapods-core/cdn_source.rb:372:in `download_and_save_with_retries_async'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-core-1.12.0/lib/cocoapods-core/cdn_source.rb:365:in `download_file_async'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-core-1.12.0/lib/cocoapods-core/cdn_source.rb:338:in `download_file'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-core-1.12.0/lib/cocoapods-core/cdn_source.rb:78:in `deprecated_local_podspecs'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-core-1.12.0/lib/cocoapods-core/cdn_source.rb:60:in `preheat_existing_files'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-core-1.12.0/lib/cocoapods-core/cdn_source.rb:257:in `update'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/sources_manager.rb:144:in `block (3 levels) in update'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/user_interface.rb:64:in `section'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/sources_manager.rb:143:in `block (2 levels) in update'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/sources_manager.rb:142:in `each'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/sources_manager.rb:142:in `block in update'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/sources_manager.rb:140:in `open'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/sources_manager.rb:140:in `update'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/installer/analyzer.rb:146:in `block in update_repositories'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/installer/analyzer.rb:144:in `each'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/installer/analyzer.rb:144:in `update_repositories'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/installer.rb:240:in `block in resolve_dependencies'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/user_interface.rb:64:in `section'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/installer.rb:239:in `resolve_dependencies'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/installer.rb:162:in `install!'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/command/install.rb:52:in `run'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/claide-1.1.0/lib/claide/command.rb:334:in `run'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/lib/cocoapods/command.rb:52:in `run'
/Users/admin/.rvm/rubies/ruby-3.2.1/lib/ruby/gems/3.2.0/gems/cocoapods-1.12.0/bin/pod:55:in `<top (required)>'
/opt/homebrew/Cellar/cocoapods/1.11.3_1/libexec/bin/pod:25:in `load'
/opt/homebrew/Cellar/cocoapods/1.11.3_1/libexec/bin/pod:25:in `<main>'
```

## Project that demonstrates the issue

https://github.com/AppLovin/AppLovin-MAX-SDK-iOS/tree/master/AppLovin%20MAX%20Demo%20App%20-%20Swift

**Comment by Gargo** [comment]
Yes, I resolved it by executing:
```
gem install ffi
```

**Comment by Dibora-Dagne** [comment]
I am getting the same error and I ran ``` gem install ffi``` no luck.

**Comment by charlieperson** [comment]
@Dibora-Dagne Did you ever figure this out?

**Comment by ahmeedev** [comment]
i'm facing this same issue. does anyone know the solution?

**Comment by pcantrell** [comment]
In my case, I did two things and I'm not sure which of them fixed it:

1. I had previously installed CocoaPods with Homebrew. I uninstalled it, then reinstalled it as a gem:
    ```bash
    brew uninstall cocoapods
    gem install cocoapods
    ```
2. I was trying to open a somewhat old project, which was locked to some old gems (possibly including an outdated ffi gem). I forced a refresh of all the bundled gems:
    ```bash
    rm Gemfile.lock
    bundle
    ```

After doing both of those things, `pod install` succeeded.
